src/rm_cms.c

```diff
@@ -25,14 +30,11 @@ static int GetCMSKey(RedisModuleCtx *ctx, RedisModuleString *keyName,
                     CMSketch **cms, int mode) {
     RedisModuleKey *key = RedisModule_OpenKey(ctx, keyName, mode);
     if(RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) {
-        RedisModule_ReplyWithError(ctx, "CMS: key does not exist");
-        return REDISMODULE_ERR;
+        INNER_ERROR("CMS: key does not exist");
```

Comment:
This will leak `key`. If you see what we do in a lot of other modules, at the end of the function we have a `done` block where we apply `goto`.

Reply:
I use RedisModule_AutoMemory so this isn't an issue.

I was very surprised to see goto but then read the one case to use it is cleanup

Reply:
`goto` isn't evil; it's specifically meant (and used) for cases like these. In any event, at least specify this in a comment.

I personally am not a fan of AutoMemory, especially in more complex commands; but I suppose cms is fairly straight forward